lib/codecs.js

```diff
@@ -0,0 +1,23 @@
+var iconv = require('iconv-lite');
+
+var cache = {};
+
+function getCodec(encoding) {
+  var codec = cache[encoding];
+  if (!!codec || !encoding || cache.hasOwnProperty(encoding)) {
+    return codec;
+  }
+
+  try {
+    codec = iconv.getCodec(encoding);
+  } catch (err) {
+    // Unsupported codec
+  }
+
+  cache[encoding] = codec;
+  return codec;
+}
+
+getCodec('utf8');
```

Comment:
Can't we just do `var cache = { utf8: iconv.getCodec('utf8') }` above since we know it will be there?